teps/0122-reproducibility-of-complete-build-instructions.md

```diff
@@ -0,0 +1,297 @@
+
+### TaskRun Specification
+
+The table below provides the name and description of the API fields that the TaskRun requires. In addition, it also lists if that information is required, not required, or provided by the provenance. If it is required, it also shows where the information should be provided. Note that since task run provides information during runtime, most of the information is contained in the **invocation.Parameters** section of the provenance.
+
+
+| | **Metadata**      | | |
+| ------ | ------------- | ----- | ----- |
+| **Field Name**    | **Description** | **Not Required / Required / Provided by provenance** | **Insert in provenance** |
+| name| Name of the task run| Required (provide an audit trail)|  |
+| | **Spec** | | |
+| resources | resources used by task | Provided (can be extracted from steps/entrypoint) |  |
+| service account name | name of the service account | Required | buildConfig |
+| params | parameter values provided by taskrun | Provided |  |
+| workspaces | workspaces used by the task | Required | buildConfig |
+| pod template | | Required | buildConfig |
+| Timeout | time in which a task should complete | Not Required (can probably use default?) |  |
+| StepOverrides | Override Step configuration specified in a Task. Currently we can override compute resources. | Not Required |  |
+| SidecarOverrides | Override Sidecar configuration specified in a Task. Currently we can override compute resources. | Not Required |  |
+| ComputeResources | Configure compute resources required by the steps in the task. | Required (because this could also indicate a minimum amount of resources required for the task run.) | buildConfig |
+| [TaskSpec](#task-specification) | Specification of the resolved task | See [spec](#task-specification) | buildConfig |
+| [TaskRef](#taskref)  | Details of the referenced task | Not Required (Since we save the complete TaskSpec even that of a remote task) |  |
+| | **Status** | | |
+| Task Results | Results produced by the task run | Not Required | buildConfig |
+
+### Tekton System Versioning
+The **version (uri and commit-sha)** of **Tekton pipelines** (that reconciled the task run), **Tekton chains** (that produced the provenance from the task run) and **remote resolvers**. Capturing this information is vital for reproducibility of the task run and can be stored in **invocation.environment** section of the provenance.
```

Comment:
Does host information belong in here? The big one is the host architecture (amd64 vs arm64). Other things like SELinux, container runtimes, and kubernetes version could also impact reproducibility.

Reply:
Good question. I wanted to focus here on Tekton related completeness.

There are lots of other things like you mentioned and probably more that need to be thought of. I can put that out-of-scope for now to capture it. WDYT?

Reply:
Sounds good. I think it would be useful if that provenance was captured, but understand if it's pushed to a separate effort.